documentation/Tutorials/WPF/CategoryEditorHookup.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: page
+title: Hook up the CategoryViewModel in the Shell
+---
+
+[Contents](Contents) [Previous](Categories1) [Next](CategoryView)
+
+## Hook up the CategoryViewModel in the Shell
+
+This is a very short step, but we now can hook the new Category UserControl into the Shell. We need two things to do:
+
++ Create a method to invoke the category editor.
++ On application start activate the CategoryViewModel and CategoryView in  the shell.
+
+### Invoke the category editor
+
+In the ``ShellViewModel`` we create a new method:
+
+```csharp
+    public async Task EditCategories()
+      {
+      var viewmodel = IoC.Get<CategoryViewModel>();
+      await ActivateItemAsync(viewmodel, new CancellationToken());
+      }
+```
+
+**IoC** stands for Inversion of Control. As you can see it is static and it will retrieve the viewmodel of the specified type from the container that holds all viewmodels. Remember, this container is created in the ``BootstrapperBase`` using reflection.
```

Comment:
You did not hook up a container in the bootstrapper, so caliburn uses a Activator to create the viewmodels in this case. So the container is not created as such.

Reply:
You need to help me out here. I agree, I did not explicitly create an IoC container. But in the ShellviewModel I use this code to get the viewmodel and then to show it:

 var viewmodel = IoC.Get<CategoryViewModel>();
 await ActivateItemAsync(viewmodel, new CancellationToken());

This refers to a static class, named IoC, which includes in the description in the source code this text:

  /// <summary>
  /// Used by the framework to pull instances from an IoC container and to inject dependencies into certain existing classes.
  /// </summary>

My modest understanding of this is, that it is a two step process, first you retrieve the ViewModel and View for a container and the you activate it. If you want to do anything more fancy, e.g. use an IWindowManager, you need to override this simple approach by creating your own container and the explicitly add instances to the container. 

So, I do not understand this comment and it may very hard to explain new users what is going on. I think I can explain this two step process, but I do not get it if you say there is no container created.

Reply:
If you look in the [BootstrapperBase](https://github.com/Caliburn-Micro/Caliburn.Micro/blob/master/src/Caliburn.Micro.Platform/Platforms/net46/Bootstrapper.cs) you can see which methods are hooked up to the `IoC` class. In the case that the `GetInstance`, `GetAllInstances` and `BuildUp` methods are not overwritten then `Activator.CreateInstance` is used. So no container is involved if the methods are not overwritten :)

Reply:
Thanks, I adapted the text. I am not sure how relevant it is for a starting user to fully understand this, but now there is a very brief explanation you can explore if you want.

Reply:
thanks for adding this